### PR TITLE
fix: handle default string value properly

### PR DESCRIPTION
**Pull request by Peeeekay**
This PR fixes the bug found recently where the default value in the starlark package was being overwritten by the "" passed by the frontend.

### CODE REVIEW

File: engine/frontend/src/component/PackageCatalogForm.js [hunk 1]
```diff
@@ -101,6 +101,20 @@ const checkValidIntType = (data) => {
...
+        return (/^-?[\d]*(\.[\d]+)?$/g).test(value);
+    }
+    if (data === "undefined") {
+        return false
+    }
+    return isValidFloat(data)
+}
+
+const checkValidBooleanType = (data) => {
+    return ["TRUE", "FALSE"].contains(data)
```
Comment (adschwartz) [inline engine/frontend/src/component/PackageCatalogForm.js]: @Peeeekay I'd recommend lowercasing the data to make sure you can match invariantly to casing

File: engine/frontend/src/component/PackageCatalogForm.js [hunk 2]
```diff
@@ -101,6 +101,20 @@ const checkValidIntType = (data) => {
     }
 }
 
+const checkValidFloatType = (data) => {
+    const  isValidFloat = (value) => {
+        return (/^-?[\d]*(\.[\d]+)?$/g).test(value);
```
Comment (adschwartz) [inline engine/frontend/src/component/PackageCatalogForm.js]: @Peeeekay It's not wrong to use a regex per se (I haven't tested the validity of this), but can we not use Number.parsefloat or [what this guy is saying](https://stackoverflow.com/a/52740766)?

File: engine/frontend/src/component/PackageCatalog.js [hunk 3]
```diff
@@ -45,12 +45,12 @@ const PackageCatalog = ({kurtosisPackages: defaultPackages}) => {
         }
         const filteredPackages = defaultPackages.filter(pack => {
                 if ("name" in pack) {
-                    return pack.name.includes(value)
+                    // lowercase everything so that it works for both cases
+                    return pack.name.toLowerCase().includes(value.toLowerCase())
```
Comment (adschwartz) [inline engine/frontend/src/component/PackageCatalog.js]: @Peeeekay maybe also add left and right trim to remove any space?

**Review by adschwartz — CHANGES_REQUESTED** [review]
Couple of comments to look at. What do you think about adding some unit tests for this to catch any obvious mistakes?

File: engine/frontend/src/component/PackageCatalogForm.js (hunk 2 above)
Comment (Peeeekay) [reply engine/frontend/src/component/PackageCatalogForm.js]: I have used the regex that was was the second answer in the slack link. IsNaN has an edge-case where "" returns true; for our initial usecases the regex seems appealing. Parsefloat also has issues where it returns true for some of failure modes. @adschwartz

Comment (adschwartz) [reply engine/frontend/src/component/PackageCatalogForm.js]: I think we can just check for an empty string no? i.e. if you get `""` then it's not a float. We should do early check for empty string null/undefined as well.

Comment (Peeeekay) [reply engine/frontend/src/component/PackageCatalogForm.js:106]: hm ok; i can do that.

**Comment by Peeeekay** [comment]
> Couple of comments to look at. What do you think about adding some unit tests for this to catch any obvious mistakes?

@adschwartz is it cool if i do that in another PR because it would require me to add some plumbing - but this can unblock nam.

File: engine/frontend/src/component/PackageCatalogForm.js (hunk 2 above)
Comment (Peeeekay) [reply engine/frontend/src/component/PackageCatalogForm.js]: oh nevermind; i was looking at different link; doing it rn

File: engine/frontend/src/component/PackageCatalog.js [hunk 4]
```diff
@@ -45,12 +45,13 @@ const PackageCatalog = ({kurtosisPackages: defaultPackages}) => {
         }
         const filteredPackages = defaultPackages.filter(pack => {
                 if ("name" in pack) {
-                    return pack.name.includes(value)
+                    // lowercase everything so that it works for both cases
+                    const trimmedValue = value.trim()
```
Comment (adschwartz) [inline engine/frontend/src/component/PackageCatalog.js:49]: @Peeeekay do we need to check for undefined and null pointer?

Comment (adschwartz) [reply engine/frontend/src/component/PackageCatalog.js:49]: Same comment other places

**Review by adschwartz — APPROVED** [review]
@Peeeekay Unblocking you but I think we should follow up with some increased quality and a unit test if possible.

File: engine/frontend/src/component/PackageCatalog.js (hunk 4 above)
Comment (Peeeekay) [reply engine/frontend/src/component/PackageCatalog.js:49]: The default value is `""`; the initial state for all the inputs is an empty string.. so it shouldn't be an issue

**Pull request merged**